dask_match/core.py

```diff
@@ -110,6 +110,10 @@ def __getattr__(self, key):
         else:
             return object.__getattribute__(self, key)
 
+    @property
+    def index(self):
+        return Index(self)
```

Comment:
I'm wondering if we want various base classes, like `FrameExpr`, `DataFrameExpr`, `SeriesExpr`, and `ScalarExpr` to make sure we only expose attributes like `index` when it makes sense.